KicadModTree/nodes/specialized/PadArray.py

```diff
@@ -231,10 +231,9 @@ def _createPads(self, **kwargs):
             delta_pos = Vector2D(0, 0)
 
         for i, number in enumerate(pad_numbers):
-            includePad = (i + self.initialPin) not in self.exclude_pin_list
-            for exi in self.exclude_pin_list:
-                if (i + self.initialPin) == exi:
-                    includePad = False
+            includePad = True
+            if type(self.initialPin) == 'int':
+                includePad = (self.initialPin + i) not in self.exclude_pin_list
```

Comment:
To be honest i would do something like the following:

```python
if number != "":
    include = number not in self.exclude_pin_list
else:
    include = i not in self.exclude_pin_list
```

As far as i can tell this feature is currently only used by the vh connectors and the ipc ic package scripts. Both should still work with my suggested modification (We need to check if `SOIC-4_4.55x3.7mm_P2.54mm` is still generated correctly after the change)

Reply:
Perfect! I will wait for https://github.com/pointhi/kicad-footprint-generator/issues/436 to be merged, then I will fix the conflicts in my PR to use your changes after everything has been tested so that there are not regressions.

Thanks!

Reply:
436 is an issue not a pull request

Reply:
I've pushed another version that I think is more complete than your proposal.

In your proposal if an str based list is passed to padarray exclusion would not be tested. The current solution checks for that.